library/src/main/java/com/schibsted/spain/barista/cleardata/ClearDatabaseRule.java

```diff
@@ -0,0 +1,112 @@
+package com.schibsted.spain.barista.cleardata;
+
+import android.content.Context;
+import android.database.Cursor;
+import android.database.sqlite.SQLiteDatabase;
+import android.database.sqlite.SQLiteException;
+import android.support.test.InstrumentationRegistry;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+
+/**
+ * This rule clears all app's Databases before running each test
+ */
+public class ClearDatabaseRule implements TestRule {
+
+  private static final String[] UNINTERESTING_FILENAME_SUFFIXES = new String[] {
+      "-journal",
+      "-shm",
+      "-uid",
+      "-wal"
+  };
+
+  @Override
+  public Statement apply(final Statement base, Description description) {
+    return new Statement() {
+      @Override
+      public void evaluate() throws Throwable {
+        clearData();
+        base.evaluate();
+        clearData();
+      }
+    };
+  }
+
+  private void clearData() {
+    for (File dbFile : tidyDatabaseList(getDatabaseFiles())) {
+      SQLiteDatabase database = performOpen(dbFile, 0);
+      for (String table : getTableNames(database)) {
+        database.delete(table, null, null);
+      }
+      database.close();
+    }
+  }
+
+  private List<File> getDatabaseFiles() {
+    Context context = InstrumentationRegistry.getTargetContext();
+    List<File> databaseFiles = new ArrayList<>();
+    for (String databaseName : context.databaseList()) {
+      databaseFiles.add(context.getDatabasePath(databaseName));
+    }
+    return databaseFiles;
+  }
+
+  private List<String> getTableNames(SQLiteDatabase database) throws SQLiteException {
+    Cursor cursor = database.rawQuery("SELECT name FROM sqlite_master WHERE type IN (?, ?)",
+        new String[] { "table", "view" });
+    try {
+      List<String> tableNames = new ArrayList<>();
+      while (cursor.moveToNext()) {
+        tableNames.add(cursor.getString(0));
+      }
+      return tableNames;
+    } finally {
+      cursor.close();
+    }
+  }
+
+  private SQLiteDatabase performOpen(File databaseFile, int options) {
+    int flags = SQLiteDatabase.OPEN_READWRITE;
+
+    return SQLiteDatabase.openDatabase(
+        databaseFile.getAbsolutePath(),
+        null /* cursorFactory */,
+        flags);
+  }
+
+  /**
+   * Attempt to smartly eliminate uninteresting shadow databases such as -journal and -uid.  Note
+   * that this only removes the database if it is true that it shadows another database lacking
+   * the uninteresting suffix.
+   *
+   * @param databaseFiles Raw list of database files.
+   * @return Tidied list with shadow databases removed.
+   */
+  private static List<File> tidyDatabaseList(List<File> databaseFiles) {
```

Comment:
This method name sounds strange... Don't know. I don't have a suggestion, by the way :·)